core/arch/arm/include/ffa.h

```diff
@@ -0,0 +1,85 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Copyright (c) 2020, Linaro Limited
+ * Copyright (c) 2018-2019, Arm Limited. All rights reserved.
+ */
+
+#ifndef __FFA_H
+#define __FFA_H
+
+/* This is based on the FF-A 1.0 EAC specification */
+
+#include <smccc.h>
+#include <stdint.h>
+
+/* Error codes */
+#define FFA_NOT_SUPPORTED	-1
+#define FFA_INVALID_PARAMETERS	-2
+#define FFA_NO_MEMORY		-3
+#define FFA_BUSY		-4
+#define FFA_INTERRUPTED		-5
+#define FFA_DENIED		-6
+#define FFA_RETRY		-7
+
+/* FFA_VERSION helpers */
+#define FFA_VERSION_MAJOR		U(1)
+#define FFA_VERSION_MAJOR_SHIFT		16
+#define FFA_VERSION_MAJOR_MASK		U(0x7FFF)
+#define FFA_VERSION_MINOR		U(0)
+#define FFA_VERSION_MINOR_SHIFT		0
+#define FFA_VERSION_MINOR_MASK		U(0xFFFF)
+#define MAKE_FFA_VERSION(major, minor)	((((major) & FFA_VERSION_MAJOR_MASK)  \
+						<< FFA_VERSION_MAJOR_SHIFT) | \
+					((minor) & FFA_VERSION_MINOR_MASK))
+
+/* FFA_MSG_SEND helpers */
+#define FFA_MSG_SEND_ATTRS_BLK_SHIFT	U(0)
+#define FFA_MSG_SEND_ATTRS_BLK_MASK	U(0x1)
```

Comment:
prefer BIT(0) as this is a bit field mask

Reply:
I'll remove these helpers, it turned out the FFA_MSG_SEND is not used in secure world.